### ISSUE TITLE
FDSN redirects from http to https are not followed

### ISSUE THREAD

**Issue by yetinam** [obspy/obspy#3210]
### Avoid duplicates

- [X] I searched existing issues

### Bug Summary

When trying to access data from NCEDC through FDSN I got an error. From what I can tell, the hard-coded URL links to the http version, however, NCEDC has moved to https only. Trying to access the http version returns a `308 Permanent Redirect` that obspy does not follow.

I think there are two ways to solve the issue:
- Simply update the url to https. However, I think the issue is bound to reappear for other providers at some point.
- Change the code to automatically follow the redirect.

For now, the easy workaround (see also code to reproduce below) is explicitly quering the URL with https.

### Code to Reproduce

```python
from obspy.clients.fdsn import Client

# Fails - Added debug flag for more details
Client("http://service.ncedc.org", debug=True)  # Equivalent to Client("NCEDC")

# Works
Client("https://service.ncedc.org")
```

### Error Traceback

```Python traceback
Installed new opener with handlers: [<obspy.clients.fdsn.client.CustomRedirectHandler object at 0x7f4737f88e20>]
Base URL: http://service.ncedc.org
Request Headers: {'User-Agent': 'ObsPy/1.3.1 (Linux-5.15.0-52-generic-x86_64-with-glibc2.31, Python 3.10.6)'}
Downloading http://service.ncedc.org/fdsnws/dataselect/1/application.wadl with requesting gzip compression
Downloading http://service.ncedc.org/fdsnws/event/1/application.wadl with requesting gzip compression
Downloading http://service.ncedc.org/fdsnws/station/1/application.wadl with requesting gzip compression
Downloading http://service.ncedc.org/fdsnws/event/1/catalogs with requesting gzip compression
Downloading http://service.ncedc.org/fdsnws/event/1/contributors with requesting gzip compression
HTTP error 308, reason Permanent Redirect, while downloading 'http://service.ncedc.org/fdsnws/event/1/catalogs': b'<!DOCTYPE HTML PUBLIC "-//IETF//DTD HTML 2.0//EN">\n<html><head>\n<title>308 Permanent Redirect</title>\n</head><body>\n<h1>Permanent Redirect</h1>\n<p>The document has moved <a href="https://service.ncedc.org/fdsnws/event/1/catalogs">here</a>.</p>\n</body></html>\n'
HTTP error 308, reason Permanent Redirect, while downloading 'http://service.ncedc.org/fdsnws/event/1/application.wadl': b'<!DOCTYPE HTML PUBLIC "-//IETF//DTD HTML 2.0//EN">\n<html><head>\n<title>308 Permanent Redirect</title>\n</head><body>\n<h1>Permanent Redirect</h1>\n<p>The document has moved <a href="https://service.ncedc.org/fdsnws/event/1/application.wadl">here</a>.</p>\n</body></html>\n'
HTTP error 308, reason Permanent Redirect, while downloading 'http://service.ncedc.org/fdsnws/station/1/application.wadl': b'<!DOCTYPE HTML PUBLIC "-//IETF//DTD HTML 2.0//EN">\n<html><head>\n<title>308 Permanent Redirect</title>\n</head><body>\n<h1>Permanent Redirect</h1>\n<p>The document has moved <a href="https://service.ncedc.org/fdsnws/station/1/application.wadl">here</a>.</p>\n</body></html>\n'
HTTP error 308, reason Permanent Redirect, while downloading 'http://service.ncedc.org/fdsnws/dataselect/1/application.wadl': b'<!DOCTYPE HTML PUBLIC "-//IETF//DTD HTML 2.0//EN">\n<html><head>\n<title>308 Permanent Redirect</title>\n</head><body>\n<h1>Permanent Redirect</h1>\n<p>The document has moved <a href="https://service.ncedc.org/fdsnws/dataselect/1/application.wadl">here</a>.</p>\n</body></html>\n'
HTTP error 308, reason Permanent Redirect, while downloading 'http://service.ncedc.org/fdsnws/event/1/contributors': b'<!DOCTYPE HTML PUBLIC "-//IETF//DTD HTML 2.0//EN">\n<html><head>\n<title>308 Permanent Redirect</title>\n</head><body>\n<h1>Permanent Redirect</h1>\n<p>The document has moved <a href="https://service.ncedc.org/fdsnws/event/1/contributors">here</a>.</p>\n</body></html>\n'
---------------------------------------------------------------------------
FDSNNoServiceException                    Traceback (most recent call last)
Cell In [4], line 1
----> 1 client = Client("http://service.ncedc.org", debug=True)

File ~/anaconda3/envs/lfe-datasets/lib/python3.10/site-packages/obspy/clients/fdsn/client.py:277, in Client.__init__(self, base_url, major_versions, user, password, user_agent, debug, timeout, service_mappings, force_redirect, eida_token, _discover_services)
    274     print("Request Headers: %s" % str(self.request_headers))
    276 if _discover_services:
--> 277     self._discover_services()
    278 else:
    279     self.services = DEFAULT_SERVICES

File ~/anaconda3/envs/lfe-datasets/lib/python3.10/site-packages/obspy/clients/fdsn/client.py:1577, in Client._discover_services(self)
   1572         raise FDSNRedirectException(", ".join(redirect_messages))
   1574     msg = ("No FDSN services could be discovered at '%s'. This could "
   1575            "be due to a temporary service outage or an invalid FDSN "
   1576            "service address." % self.base_url)
-> 1577     raise FDSNNoServiceException(msg)
   1578 # Cache.
   1579 if self.debug is True:

FDSNNoServiceException: No FDSN services could be discovered at 'http://service.ncedc.org'. This could be due to a temporary service outage or an invalid FDSN service address.
```

### ObsPy Version?

1.3.1

### Operating System?

Ubuntu

### Python Version?

3.10

### Installation Method?

pip

**Comment by ThomasLecocq** [comment]
solved with #3201 and related PR.

**Comment by yetinam** [comment]
Sorry for the double post. I was obviously not thorough enough when checking for duplicates. And thanks for the quick reply.
